apis/v1alpha1/certificate.go

```diff
@@ -0,0 +1,143 @@
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CertificateSpec defines the desired state of Certificate.
+type CertificateSpec struct {
+
+	// The Amazon Resource Name (ARN) of the private certificate authority (CA)
+	// that will be used to issue the certificate. If you do not provide an ARN
+	// and you are trying to request a private certificate, ACM will attempt to
+	// issue a public certificate. For more information about private CAs, see the
+	// Amazon Web Services Private Certificate Authority (https://docs.aws.amazon.com/privateca/latest/userguide/PcaWelcome.html)
+	// user guide. The ARN must have the following form:
+	//
+	// arn:aws:acm-pca:region:account:certificate-authority/12345678-1234-1234-1234-123456789012
+	CertificateAuthorityARN *string `json:"certificateAuthorityARN,omitempty"`
+	// Fully qualified domain name (FQDN), such as www.example.com, that you want
+	// to secure with an ACM certificate. Use an asterisk (*) to create a wildcard
+	// certificate that protects several sites in the same domain. For example,
+	// *.example.com protects www.example.com, site.example.com, and images.example.com.
+	//
+	// In compliance with RFC 5280 (https://datatracker.ietf.org/doc/html/rfc5280),
+	// the length of the domain name (technically, the Common Name) that you provide
+	// cannot exceed 64 octets (characters), including periods. To add a longer
+	// domain name, specify it in the Subject Alternative Name field, which supports
+	// names up to 253 octets in length.
+	// +kubebuilder:validation:Required
+	DomainName *string `json:"domainName"`
+	// The domain name that you want ACM to use to send you emails so that you can
+	// validate domain ownership.
+	DomainValidationOptions []*DomainValidationOption `json:"domainValidationOptions,omitempty"`
+	// Customer chosen string that can be used to distinguish between calls to RequestCertificate.
+	// Idempotency tokens time out after one hour. Therefore, if you call RequestCertificate
+	// multiple times with the same idempotency token within one hour, ACM recognizes
+	// that you are requesting only one certificate and will issue only one. If
+	// you change the idempotency token for each call, ACM recognizes that you are
+	// requesting multiple certificates.
+	IDempotencyToken *string `json:"idempotencyToken,omitempty"`
```

Comment:
Oops looks like we have an initialism fail

Reply:
Actually, I'm going to remove that field from the Certificate resource. It's not needed and is an artifact of the weird ACM APIs.